capture/capture.proto

```diff
@@ -175,12 +175,16 @@ message CaptureResponse {
 
   // Commit previous captured documents.
   message Commit {
-    // Optional driver checkpoint of this transaction.
-    // If provided, the most recent checkpoint will be persisted by the
-    // Flow runtime and returned in a future CaptureRequest.
-    bytes driver_checkpoint_json = 1 [
+    // Optional driver checkpoint update of this transaction,
+    // applied as an RFC7396 Merge Patch to the driver checkpoint
+    // of the immediately preceeding transaction (or to an empty
+    // JSON object `{}` if there is no preceeding transaction).
+    //
+    // The patched checkpoint will be persisted by the Flow
+    // runtime and returned in a future CaptureRequest.
+    bytes driver_checkpoint_merge_patch_json = 1 [
       (gogoproto.casttype) = "encoding/json.RawMessage",
-      json_name = "driverCheckpoint"
+      json_name = "driverCheckpointMergePatch"
```

Comment:
The idea here is that a full checkpoint will "patch" into any previous state with by overwriting everything, which while redundant, should still be correct, right?

Is there ever a circumstance that we will need to differentiate actual patches from full state checkpoints?